### PR TITLE
Started using direct ByteBuffers for our performance tests

### PR DESCRIPTION
**Pull request by gaaclarke**
This takes advantage of our work to speed up platform channels.  It's expected this will drop the times of the 1MB binary test down ~45%.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/benchmarks/platform_channels_benchmarks/android/app/src/main/kotlin/com/example/platform_channels_benchmarks/MainActivity.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ class MainActivity: FlutterActivity() {
         } }
         val basicStandard = BasicMessageChannel(flutterEngine.dartExecutor, "dev.flutter.echo.basic.standard", StandardMessageCodec.INSTANCE)
         basicStandard.setMessageHandler { message, reply -> reply.reply(message) }
-        val basicBinary = BasicMessageChannel(flutterEngine.dartExecutor, "dev.flutter.echo.basic.binary", BinaryCodec.INSTANCE)
+        val basicBinary = BasicMessageChannel(flutterEngine.dartExecutor, "dev.flutter.echo.basic.binary", BinaryCodec.INSTANCE_DIRECT)
```
Comment (goderbauer) [inline dev/benchmarks/platform_channels_benchmarks/android/app/src/main/kotlin/com/example/platform_channels_benchmarks/MainActivity.kt:26]: I tried to find `INSTANCE_DIRECT` in our API docs to see what the difference to `INSTANCE` is, but wasn't able to locate it. Shouldn't it appear here:

https://master-api.flutter.dev/javadoc/io/flutter/plugin/common/BinaryCodec.html

Comment (gaaclarke) [reply dev/benchmarks/platform_channels_benchmarks/android/app/src/main/kotlin/com/example/platform_channels_benchmarks/MainActivity.kt:26]: It got reverted here https://github.com/flutter/engine/pull/26470.  I'm working on the reland right now.

**Comment by gaaclarke** [comment]
Don't merge until https://github.com/flutter/engine/pull/26515 has rolled into the framework.

**Comment by gaaclarke** [comment]
The roll happened in https://github.com/flutter/flutter/pull/83742, this is good to merge.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**